src/box/xrow_update_array.c

```diff
@@ -320,7 +320,8 @@ xrow_update_array_append_nils(struct xrow_update_field *field,
 {
 	struct xrow_update_rope *rope = field->array.rope;
 	uint32_t size = xrow_update_rope_size(rope);
-	if (op->field_no < 0 || (uint32_t)op->field_no <= size)
+	if (op->lexer.src != NULL || op->field_no < 0 ||
```

Comment:
But nothing changed. You still check src != NULL even for perfectly normal operations. The idea was that you should not check it unless you see the field_no > size.